File: AUGUST-2025/03-08-2025/p2106.dart
class Solution {
  int maxTotalFruits(List<List<int>> fruits, int startPos, int k) {
    int n = fruits.length;
    int maxFruits = 0, windowSum = 0;
    int left = 0;

    for (int right = 0; right < n; right++) {
      windowSum += fruits[right][1];

      while (left <= right) {
        int leftPos = fruits[left][0];
        int rightPos = fruits[right][0];

        int directLeft = (startPos - leftPos).abs();
        int directRight = (startPos - rightPos).abs();
        int totalWidth = rightPos - leftPos;
        int minCost = (directLeft < directRight ? directLeft : directRight) + totalWidth;

        if (minCost <= k) break;

        windowSum -= fruits[left][1];
        left++;
      }

      if (windowSum > maxFruits) maxFruits = windowSum;
    }

    return maxFruits;
  }
}

File: AUGUST-2025/03-08-2025/p2106.rs
impl Solution {
    pub fn max_total_fruits(f: Vec<Vec<i32>>, sp: i32, k: i32) -> i32 {
        let (mut q, mut s) = (0, 0);
        f.iter().map(|v| {
            while (sp + v[0] - 2*f[q][0]).min(2*v[0] - sp - f[q][0]) > k { s -= f[q][1]; q += 1 }
            if (v[0] - sp).abs() > k { q += 1; 0 } else { s += v[1]; s }
        }).max().unwrap()
    }

}